### ISSUE TITLE
Merge security code from UCLA

### ISSUE THREAD

**Issue by tntech-ngin** [tntech-ngin/ndn-hydra#11]
Sorry for the late email. I promised I will send an email to describe what I have done for hydra security. Here's the description of my code:

1/ envelope: https://github.com/tianyuan129/trust-info-base. This is our group's latest trust schema API at last ICN. I used this to secure SVS and hydra.

2/ ndn-python-svs: https://github.com/tianyuan129/ndn-python-svs. This is my fork of Justins's python-svs.

3/ hydra: https://github.com/tianyuan129/ndn-hydra. This is my fork of hydra. Not only about security, but also I fixed some bugs in the master branch. In the examples I have a python script to bootstrap nodes (i.e., trust anchor, trust schema, certificate).

I know how to run my code directly on hosts. Should I do it on fabric hosts?

Best,
Tianyuan

**Comment by GportDev** [comment]
@tianyuan129 I'm trying to publish the svs package in PyPi, but it has a dependency on your _envelope_ package. Could you please publish this into PyPi so we can use it?

PyPi blocks the upload of packages that depend on other packages that are not in PyPi.

**Comment by tianyuan129** [comment]
Hope my part for this issue already resolved.

**Comment by GportDev** [comment]
Yes, If you can review the PR one more time would be great, I added some changes this morning.
